pyanaconda/keyboard.py

```diff
@@ -55,6 +56,39 @@ class InvalidLayoutVariantSpec(Exception):
     pass
 
 
+def _is_xwayland():
+    """Is Anaconda running in XWayland environment?
+
+    This can't be easily detected from the Anaconda because Anaconda is running as XWayland app.
+    Use xisxwayland tool for the detection.
+    """
+    try:
+        rc = execWithRedirect('xisxwayland', [])
+
+        if rc == 0:
+            return True
+        else:
+            log.debug("xisxwayland returned %s code, see xisxwayland --help for more info", rc)
+    except FileNotFoundError:
+        log.warning("The xisxwayland tool is not available! "
+                    "Taking the environment as not Wayland.")
+
+    return False
+
+
+def can_configure_keyboard():
+    """Can we configure the keyboard?
+
+    TODO: The is_wayland logic is not part of the configuration so we would have to add it to the
+    configuration otherwise it won't be accessible in the Anaconda modules.
+    This is more a temporary workaround which will be replaced in the future.
+    """
+    if _is_xwayland() and conf.system._is_live_os:
```

Comment:
I guess we could have something like `conf.system.can_run_wayland` to avoid the private property.

Reply:
It is cheaper to check these conditions in reverse order.